draft-nro-bulk-rdap.md

```diff
@@ -48,184 +48,294 @@ feature of this service.
+      "type": "application/rdap+json"
     },
+    ...
+  ],
+  "entities":
+  [
     {
-      "objectClassName": "ip network",
-      "handle": "YYYY-RIR",
-      "startAddress": "2001:db8:1::",
-      "endAddress": "2001:db8:1:ffff:ffff:ffff:ffff:ffff",
-      "ipVersion": "v6",
-      ...
+      "objectClassName": "entity",
+      "handle": "AAAA-ARIN",
+      "roles": [ "administrative", ... ]
       "links":
       [
         {
-          "value": "https://example.net/ip/2001:db8:1::/48",
+          "value": "https://example.net/ip/2001:db8::/48",
           "rel": "self",
-          "href": "https://example.net/ip/2001:db8:1::/48",
+          "href": "https://example.net/entity/AAAA-ARIN",
           "type": "application/rdap+json"
-        },
-        ...
-      ],
-      "entities":
+        }
+      ]
+    },
+    ...
+  ]
+}
+...
+{
+  "rdapConformance": [ "rdap_level_0", "nro_rdap_profile_0", "nroBulkRdap1", ... ],
+  "objectClassName": "autnum",
+  "handle": "ZZZZ-ARIN",
+  "startAutnum": 65536,
+  "endAutnum": 65541,
+  ...
+  "links":
+  [
+    {
+      "value": "https://example.net/autnum/65536",
+      "rel": "self",
+      "href": "https://example.net/autnum/65536",
+      "type": "application/rdap+json"
+    },
+    ...
+  ],
+  "entities":
+  [
+    {
+      "objectClassName": "entity",
+      "handle": "BBBB-ARIN",
+      "roles": [ "technical", ... ]
+      "links":
       [
         {
-          "objectClassName": "entity",
-          "handle": "BBBB-RIR",
-          "roles": [ "technical", ... ]
-          "links":
-          [
-            {
-              "value": "https://example.net/ip/2001:db8:1::/48",
-              "rel": "self",
-              "href": "https://example.net/entity/BBBB-RIR",
-              "type": "application/rdap+json"
-            }
-          ],
-        },
-        ...
-      ],
+          "value": "https://example.net/autnum/65536",
+          "rel": "self",
+          "href": "https://example.net/entity/BBBB-ARIN",
+          "type": "application/rdap+json"
+        }
+      ]
     },
     ...
   ]
 }
+...
 ```
 
-_Figure 1_
-
 # Extension Identifier
 
-The "nroBulkRdap1" extension identifier (see (#rdap_extensions_registry)) MUST be included in the top-level
-"rdapConformance" member of the JSON object for bulk data, to signal adherence to this specification.
+The "nroBulkRdap1" extension identifier (see (#rdap_extensions_registry)) MUST be included as the value for the
+"extensionId" member in the metadata object of a bulk data response.
 
-It is RECOMMENDED that the extension identifier for the NRO RDAP Profile ("nro_rdap_profile_0") also be included in the
-top-level "rdapConformance" member. But, when in conflict, the Bulk RDAP extension requirements SHOULD supersede the NRO
-RDAP Profile extension requirements, primarily to afford bulk data compactness.
+The "rdapConformance" array for each returned data object MUST include both the "nroBulkRdap1" extension identifier and
```

Comment:
I'm not sure that the bulk context has any effect on the individual objects themselves such that this would be useful information to have at that level, so why do the objects need to include that identifier?

Reply:
Thought of including it since the data compactness requirements that would supersede the NRO RDAP Profile requirements are because of it.

Reply:
That's a good point, thanks.  I think in that case, some text like "clients who host this data should omit the nroBulkRdap1 extension identifier when including objects in responses in non-bulk contexts" would be useful.

Reply:
Could you please elaborate on a "non-bulk context"?

Reply:
For example, a bulk RDAP client retrieves a snapshot, and then uses that data to host a mirror server for local use.  When it returns an object for a standard query in that context, it should remove the nroBulkRdap1 extension identifier.

Reply:
Got it.